Added support to access native posix flash via host file system (#16467)
zephyrproject-rtos/zephyr: boards/posix/native_posix/doc/index.rst

  configuration. In case the file does not exists the driver will take care of
  creating the file, else the existing file is used.

>   If desired, the flash content can be accessed from the host system, for more

Reviewer: How about:
```
  The flash content can be accessed from the host system, as explained
  in the `Host based flash access`_ section.
```

  A backend/"bottom" for Zephyr's CTF tracing subsystem which writes the tracing
  data to a file in the host filesystem.
  More information can be found in :ref:`Common Tracing Format <ctf>`

Host based flash access
***********************

> In case a flash device is present, the file system partitions on the flash

Reviewer: ```suggestion
If a flash device is present, the file system partitions on the flash
```

device can be exposed through the host file system by enabling
:option:`CONFIG_POSIX_FUSE_FLASH_ACCESS`. This option enables a FUSE
(File system in User space) layer that maps the Zephyr file system calls to

> the required UNIX file system calls and as such allows access to flash file

Reviewer: ```suggestion
the required UNIX file system calls, and provides access to the flash file
```

> system partitions with generic operating system tools (cd, ls, mkdir, ...).

Reviewer: ```suggestion
system partitions with normal operating system commands such as
``cd``, ``ls``, and ``mkdir``.
```

By default the partitions are exposed through the directory *flash* in the
current working directory. This directory can be changed via the command line

> option *--flash-mount*. As this directory operates as mount point for FUSE

Reviewer: ```suggestion
option *--flash-mount*. As this directory operates as a mount point for FUSE,
```

you have to ensure that it exists before starting the native POSIX board.

> On exit the native POSIX board application will take care of unmounting the

Reviewer: ```suggestion
On exit, the native POSIX board application will take care of unmounting the
```

directory. In the unfortunate case that the native POSIX board application

> would crash you can cleanup the stale mount point via the program fusermount,

Reviewer: ```suggestion
crashes, you can cleanup the stale mount point by using the program ``fusermount``::
```
and delete the ``eg::`` on the next line.

  A backend/"bottom" for Zephyr's CTF tracing subsystem which writes the tracing
  data to a file in the host filesystem.
  More information can be found in :ref:`Common Tracing Format <ctf>`

Host based flash access
***********************

If a flash device is present, the file system partitions on the flash
device can be exposed through the host file system by enabling
:option:`CONFIG_FUSE_FS_ACCESS`. This option enables a FUSE
(File system in User space) layer that maps the Zephyr file system calls to
the required UNIX file system calls, and provides access to the flash file
system partitions with normal operating system commands such as ``cd``,

>  ``ls`` and ``mkdir``.

Reviewer 2: ``doc/_build/rst/doc/boards/posix/native_posix/doc/index.rst:642: WARNING: Unexpected indentation.``

Reviewer 2: @vanwinkeljan This is how it looks: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/16467/docs/boards/posix/native_posix/doc/index.html#host-based-flash-access

Reviewer: P.S.  using ``make htmldocs-fast`` avoids creating all the options documentation details and runs in a quarter of the time as the full doc build.

Author: @dbkinder is this a new make target, as I get `No rule to make target`?

>  ``ls`` and ``mkdir``.

Reviewer: There's an extra space at the front of this line (642) that you should delete :)
